pandas/core/interval.py

```diff
@@ -0,0 +1,521 @@
+        if not self.is_non_overlapping_monotonic:
+            raise KeyError('can only get slices from an IntervalIndex if '
+                           'bounds are non-overlapping and all monotonic '
+                           'increasing or decreasing')
+
+        if isinstance(label, IntervalMixin):
+            raise NotImplementedError
+
+        if ((side == 'left' and self.left.is_monotonic_increasing) or
+                (side == 'right' and self.left.is_monotonic_decreasing)):
+            sub_idx = self.right
+            if self.open_right or exclude_label:
+                label = _get_next_label(label)
+        else:
+            sub_idx = self.left
+            if self.open_left or exclude_label:
+                label = _get_prev_label(label)
+
+        return sub_idx._searchsorted_monotonic(label, side)
+
+    def _get_loc_only_exact_matches(self, key):
+        return self._multiindex._tuple_index.get_loc(key)
+
+    def _find_non_overlapping_monotonic_bounds(self, key):
+        if isinstance(key, IntervalMixin):
+            start = self._searchsorted_monotonic(
+                key.left, 'left', exclude_label=key.open_left)
+            stop = self._searchsorted_monotonic(
+                key.right, 'right', exclude_label=key.open_right)
+        else:
+            # scalar
+            start = self._searchsorted_monotonic(key, 'left')
+            stop = self._searchsorted_monotonic(key, 'right')
+        return start, stop
+
+    def get_loc(self, key, method=None):
+        self._check_method(method)
+
+        original_key = key
+
+        if self.is_non_overlapping_monotonic:
+            if isinstance(key, Interval):
+                left = self._maybe_cast_slice_bound(key.left, 'left', None)
+                right = self._maybe_cast_slice_bound(key.right, 'right', None)
+                key = Interval(left, right, key.closed)
+            else:
+                key = self._maybe_cast_slice_bound(key, 'left', None)
+
+            start, stop = self._find_non_overlapping_monotonic_bounds(key)
+
+            if start + 1 == stop:
+                return start
+            elif start < stop:
+                return slice(start, stop)
+            else:
+                raise KeyError(original_key)
+
+        else:
+            # use the interval tree
+            if isinstance(key, Interval):
+                left, right = _get_interval_closed_bounds(key)
+                return self._tree.get_loc_interval(left, right)
+            else:
+                return self._tree.get_loc(key)
+
+    def get_value(self, series, key):
+        # this method seems necessary for Series.__getitem__ but I have no idea
+        # what it should actually do here...
+        loc = self.get_loc(key)  # nb. this can't handle slice objects
+        return series.iloc[loc]
+
+    def get_indexer(self, target, method=None, limit=None, tolerance=None):
+        self._check_method(method)
```

Comment:
What is our strategy for docstrings for methods that are redundant with the super class method? I guess I could just copy those over (e.g., `get_indexer.__doc__ = Index.get_indexer.__doc__`).